crates/libcgroups/src/systemd/manager.rs

```diff
@@ -353,6 +353,10 @@ impl CgroupManager for Manager {
         if pid.as_raw() == -1 {
             return Ok(());
         }
+        if self.client.transient_unit_exists(&self.unit_name) {
+            tracing::debug!("Transient unit {:?} already exists", self.unit_name);
+            return Ok(());
+        }
```

Comment:
I'm not sure if this  is correct. For exec, the unit will exist (as the container is running) , so we should probably add the pid of the new exec process into it? If so then in case of tenant container, we need a call to set unit props with pid modified? @utam0k may I ask you to take a look at this?

Reply:
In terms of OCI Runtime spec, such a behavior isn't defined. I'm not an expert on systemd, but anyway, we have to put a tenant pid into a cgroup which is the same cgroup of running container pid.

Reply:
Hey, fixed this. Now the add_task will either create a new cgroup with the provided pid, or if cgroup already exists, it will add the pid into that. I have verified it using systemd cgroup impl, and changed for cgroups-c2 , but not sure how to verify that, as even docker uses systemd cgroups.